### PR TITLE
Update Version.java

### PR DESCRIPTION
**Pull request by portlek**

### CODE REVIEW

**Comment by skuzzle** [comment]
Can you tell me which particular problem this solves? Thinking about it right now, depending on any kind of default locale could be problematic. I believe it would be better to provide an overload of `toUpperCase` taking a `Locale` as input.

Not sure whether I'll integrate this, but if so:
* wildcard import should be remove
* test case(s) for new/changed behavior must be added

**Comment by portlek** [comment]
@skuzzle 
it causes problem with Turkish letters, for example we have `I` letter, if you do toLowerCase without Locale.ROOT it will be `ı` which there is no letter in english. or another one is `i`, if you do toUpperCase without Locale.ROOT, it will be `İ` which  again there is no letter in english. so this solves the issue for users who Turkish or other users who have these letter or something like that.

**Comment by skuzzle** [comment]
I see. I still think that it would be better to provide an overload of `toUpperCase` and `toLowerCase` taking a `Locale` as parameter.

Also, I'm seriously thinking about ditching the `toUpperCase` and `toLowerCase` altogehter in favor of manually doing it like in the following snippet:

```java
final Version version = Version.create(1, 2, 3, "snapshot", "build-123");
final Version versionUpper = version.withPreRelease(version.getPreRelease().toUpperCase(Locale.ROOT));
```
It is not as convenient but gives you full control over how to upper/lower case the version. Otherwise we will sooner or later end up with adding all the methods from the `String` class to the `Version` class.

File: src/main/java/de/skuzzle/semantic/Version.java [hunk 1]
```diff
@@ -25,10 +25,7 @@
 
 import java.io.ObjectStreamException;
 import java.io.Serializable;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Comparator;
-import java.util.List;
+import java.util.*;
```
Comment (skuzzle) [inline src/main/java/de/skuzzle/semantic/Version.java:28]: Revert the wildcard import
```suggestion
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
import java.util.List;
```

**Comment by skuzzle** [comment]
Ahh, I understand the problem now. Using the default locale may indeed produce invalid (in terms of the specification) versions because only `[a-zA-Z]` are allowed as characters.
Here is a test case to reproduce it:

Add junit-pioneer:
```xml
        <dependency>
            <groupId>org.junit-pioneer</groupId>
            <artifactId>junit-pioneer</artifactId>
            <version>1.4.2</version>
            <scope>test</scope>
        </dependency>
```
Then add these tests to `VersionTest.java`:
```java
    // This would have produced an illegal identifier when using toLowerCase with default
    // locale
    // https://github.com/skuzzle/semantic-version/pull/5
    @Test
    @DefaultLocale("tr-TR")
    void testToLowerCaseWithTurkishLocale() throws Exception {
        final Version lowerCase = Version.create(1, 3, 4).withPreRelease("I")
                .toLowerCase();
        Version.create(1, 3, 4, lowerCase.getPreRelease());
    }

    // This would have produced an illegal identifier when using toLowerCase with default
    // locale
    // https://github.com/skuzzle/semantic-version/pull/5
    @Test
    @DefaultLocale("tr-TR")
    void testToUpperCaseWithTurkishLocale() throws Exception {
        final Version lowerCase = Version.create(1, 3, 4)
                .withPreRelease("i")
                .toUpperCase();
        Version.create(1, 3, 4, lowerCase.getPreRelease());
    }
```

I'm willing to accept the PR, but
* Please rebase your dev branch onto this repository's `dev` branch
* Add the above test case
* Revert the wildcard import back to explicit imports (see suggestion)

**Review by skuzzle — CHANGES_REQUESTED** [review]

**Comment by skuzzle** [comment]
Thanks a lot for your contribution. As this is quite a major bug and I haven't heard back, I pushed the changes myself now

**Pull request closed without merging**